### DISCUSSION TITLE
web support for plugin firebase_dynamic_links

### DISCUSSION THREAD

**Discussion by jakusb** [firebase/flutterfire#7238] [Feature request]
We have build an app that has a web landing page where users land anonymously to report an issue.
We would like to use dynamic links to offer the anonymous user te continue the session in the iOS or Android version of the app as a registered user.

In order to relay any information from the anonymous session (uid) we need to build the dynamic link programatically.
To our surprise this is not supported, while this seems to be the most important feature of dynamic links in flutter.
We tried but get this error:
Error: MissingPluginException(No implementation found for method FirebaseDynamicLinks#getInitialLink on channel plugins.flutter.io/firebase_dynamic_links)

What is so difficult to not support this in flutter web?
It should simply be constructing a simple url from some provided info.

Please add web support for this firebase_dynamic_links 

`                final DynamicLinkParameters parameters = DynamicLinkParameters(
                  uriPrefix: 'https://link.xxx.com/link',
                  link: Uri.parse('https://xxxx.com/app/login?uid=12345'),
                  androidParameters: AndroidParameters(
                    packageName: 'xx.xxxx.xxx_app',
                    minimumVersion: 27,
                  ),
                  iosParameters: IosParameters(
                    bundleId: 'xx.xxxx.xxx_app',
                    minimumVersion: '1.0.0',
                    appStoreId: 'xxxx',
                  ),
                  googleAnalyticsParameters: GoogleAnalyticsParameters(
                    campaign: 'reporter2user',
                    medium: 'webreporter',
                    source: 'webversion',
                  ),
                  // itunesConnectAnalyticsParameters: ItunesConnectAnalyticsParameters(
                  //   providerToken: '123456',
                  //   campaignToken: 'example-promo',
                  // ),
                  socialMetaTagParameters:  SocialMetaTagParameters(
                    title: 'Dynamic Link from Web to App',
                    description: 'This link works whether app is installed or not!',
                  ),
                );

                final Uri dynamicUrl = await parameters.buildUrl();
`

**Comment by jakusb** [comment]
I understand currently there is detection implemented to discover which Uri to build (iOS of Android), which does not make sense in a web environment.
But this could be resolved, by adding an optional argument to have the Uri build specifically for AppStore or PlayStore.

**Comment by mklepaczko** [comment]
Any feedback on that? Any plans for web support?

**Comment by jonmountjoy** [comment]
The same can be said about MacOS, which is also not supported.  I'm creating a cross platform app in Flutter, and I want to create a link I can distribute across all the platforms.

**Comment by frankuserconnect** [comment]
Any updates on this yet?

**Comment by Vaytrex** [comment]
For anyone landing here after the Aug 2025 shutdown: there's no drop-in. You need a URL that branches on User-Agent (App Store / intent:// with browser_fallback_url / website) and skips redirecting crawlers so previews survive. Full write-up with the intent format: https://linkbah.app/firebase-dynamic-links-alternative
